### DISCUSSION TITLE
How to check overall deployment status using SDK

### DISCUSSION THREAD

**Discussion by kangtao** [kubernetes-client/csharp#873] [Q&A]
Hi,

I am using this C# SDK to deploy my yaml which contains different types of resources like services, secrets, deployment, etc. It is working fine. But what is the best approach to check the overall deployment status after deployment using the SDK? I want to make sure all my pods/services/ingress are running after I deploy the yaml using SDK.

Thanks

**Comment by tg123** [comment]
kubernetes has 2 status: desired status, on the way to desired status
are you looking for how to query status of those resource?

**Comment by kangtao** [comment]
> kubernetes has 2 status: desired status, on the way to desired status
> are you looking for how to query status of those resource?

Let's say I have a web application hosted in kubernetes where I have a web service and a database service. How to determine if they are actually up after I deploy my yaml?  Do I need to query the status of these two services, or deployments, or even pods? If they all succeed, everything will be in ready or running state. But let's say somehow my web service cannot connect to my database during server starts up, and the deployment may not be in ready state and pod may be stuck in restarting.  If I call get deployments or get pods, deployments will not be ready state and pod may be in restarting state. Can I simply say my deployment fails from these status? If not, how can I determine? Thanks for your help

**Comment by brendandburns** [comment]
This is a generic Kubernetes question that is outside of the scope of this library.

There are lots of resources online for learning about Kubernetes, this doc here:

https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/

Is a pretty good place to start.

I'm going to close this issue since it is outside of the scope of this library. Please use `/reopen` or open another issue if you need assistance with this client library.

**Comment by tg123** [comment]
i guess you are talking about doing 'kubectl rollout' with the library
